src/crl.rs

```diff
@@ -52,6 +52,86 @@ pub struct CertRevocationList<'a> {
     pub crl_number: Option<&'a [u8]>,
 }
 
+/// Representation of a RFC 5280[^1] profile Certificate Revocation List (CRL) revoked certificate
+/// entry.
+///
+/// [^1]: <https://www.rfc-editor.org/rfc/rfc5280#section-5>
+pub struct RevokedCert<'a> {
+    /// Serial number of the revoked certificate.
+    pub serial_number: &'a [u8],
+
+    /// The date at which the CA processed the revocation.
+    pub revocation_date: Time,
+
+    /// Identifies the reason for the certificate revocation.
+    pub reason_code: Option<RevocationReason>,
+
+    /// Provides the date on which it is known or suspected that the private key was compromised or
+    /// that the certificate otherwise became invalid. This date may be earlier than the revocation
+    /// date which is the date at which the CA processed the revocation.
+    pub invalidity_date: Option<Time>,
+}
+
+/// Identifies the reason a certificate was revoked.
+/// See RFC 5280 §5.3.1[^1]
+///
+/// [^1] <https://www.rfc-editor.org/rfc/rfc5280#section-5.3.1>
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[allow(missing_docs)] // Not much to add above the code name.
+pub enum RevocationReason {
+    /// Unspecified should not be used, and is instead assumed by the absence of a RevocationReason
+    /// extension.
+    Unspecified,
+    KeyCompromise,
+    CaCompromise,
+    AffiliationChanged,
+    Superseded,
+    CessationOfOperation,
+    CertificateHold,
+    /// RemoveFromCrl only appears in delta CRLs that are unsupported.
+    RemoveFromCrl,
+    PrivilegeWithdrawn,
+    AaCompromise,
+}
+
+#[derive(Clone, Copy)]
+/// Controls the behaviour of [iter_revoked_certs].
+pub enum RevokedCertIteration<T> {
+    /// Continue iterating revoked certificates.
+    KeepGoing,
+
+    /// Stop iteration of revoked certificates, yielding a result.
+    Stop(Result<T, Error>),
+}
+
+impl<'a> CertRevocationList<'a> {
+    /// Iterate through the certificate revocation list's revoked certificates (if any).
```

Comment:
This docstring seems to want to evoke brackets to point at items, but I think rustdoc requires the use of backticks (like ```[`RevokedCert`]```?). Also it would be nice to use them consistently, including for the argument name `f`.

Reply:
This fell away when I reworked with the iterator, but I made sure the new `find_serial` helper has consistent rustdoc linking.